spring-cloud-sleuth-zipkin/src/main/java/org/springframework/cloud/sleuth/zipkin/ServerPropertiesEndpointLocator.java

```diff
@@ -73,11 +78,16 @@ private Integer getPort() {
 	}
 
 	private int getAddress() {
-		if (this.serverProperties!=null && this.serverProperties.getAddress() != null) {
-			return InetUtils.getIpAddressAsInt(this.serverProperties.getAddress().getHostAddress());
+		if (this.serverProperties != null && this.serverProperties.getAddress() != null) {
+			return ByteBuffer.wrap(this.serverProperties.getAddress().getAddress())
+					.getInt();
 		}
 		else {
-			return 127 << 24 | 1;
+			return ByteBuffer.wrap(getFirstNonLoopbackAddress().getAddress()).getInt();
 		}
 	}
+
+	private InetAddress getFirstNonLoopbackAddress() {
```

Comment:
inline this as it isn't more clear than calling the method findFirstNonLoopbackAddress directly